src/test/java/org/junitpioneer/jupiter/resource/TemporaryDirectoryAssertions.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2016-2022 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter.resource;
+
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+class TemporaryDirectoryAssertions {
+
+	public static final Path ROOT_TEMP_DIR = Paths.get(System.getProperty("java.io.tmpdir"));
+
+	static void assertCanAddAndReadTextFile(Path tempDir) {
+		try {
+			Path testFile = Files.createTempFile(tempDir, "some-test-file", ".txt");
+			Files.write(testFile, singletonList("some-text"));
+			assertThat(Files.readAllLines(testFile)).containsExactly("some-text");
+		}
+		catch (IOException e) {
+			fail(e);
+		}
+	}
+
+	static void assertReadableWriteableTemporaryDirectory(Path tempDir) {
```

Comment:
As my previous comment, this should not look like this, in my opinion.
My recommendation:
```java
assertThat(tempDir)
    .isTemporaryDirectory()
    .isReadable()
    .isWritable();
```

Applies to `assertEmptyReadableWriteableTemporaryDirectory` as well.

Reply:
Sounds reasonable to me. Fixed in .

As part of that commit, I've elected to completely remove `TemporaryDirectoryAssertions` and move its sole remaining member, `ROOT_TEMP_DIR`, into its most frequent user, `TemporaryDirectoryTests`. For your information, the only other user of `ROOT_TEMP_DIR` is `TemporaryDirectoryDirTests`, which uses it only once.

Let me know if you think I should move `ROOT_TEMP_DIR` into its own file.